### ISSUE TITLE
Strange broken symbol names tracing recent v8

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#344]
Hi, there's some problem on displaying symbol name when tracing v8.
```
# DURATION     TID     FUNCTION
            [  5341] | <55b354f62ec0>() {
            [  5341] |   <55b354f60940>() {
            [  5341] |     <55b35504d150>() {
            [  5341] |       <55b35504d1b0>() {
            [  5341] |         v8::internal::AstNodeFactory::NewEmptyStatement() {
            [  5341] |           v8::internal::UnwrapSharedFunctionInfoFromJSValue() {
            [  5341] |             v8::internal::ParserBase::ParseClassDeclaration() {
  51.021 ms [  5341] |               v8::internal::ScopeInfo::StackSlotCount();
        ...
```
Here is sid map:
```
55b35496a000-55b354f23000 r--p 00000000 08:11 63059744      /home/honggyu/v8/out/x64.release/d8
55b354f23000-55b356eff000 r-xp 005b9000 08:11 63059744      /home/honggyu/v8/out/x64.release/d8
55b356eff000-55b356f05000 rw-p 02595000 08:11 63059744      /home/honggyu/v8/out/x64.release/d8
55b356f05000-55b356ffa000 r--p 0259b000 08:11 63059744      /home/honggyu/v8/out/x64.release/d8
55b356ffa000-55b357018000 rw-p 00000000 00:00 0
55b358486000-55b3584a7000 rw-p 00000000 00:00 0             [heap]
        ...
```
d8.sym file contains `main` function in its symbol list as follows:
```
00000000005f8c20 w _ZNSt3__110unique_ptrIN12v8_inspector11V8Inspector7ChannelENS_14default_deleteIS3_EEE5resetEPS3_
00000000005f8d40 t _ZN2v812_GLOBAL__N_124ArrayBufferAllocatorBaseD2Ev
00000000005f8ec0 T main
00000000005f8f10 w _ZN2v811SourceGroup13IsolateThreadD0Ev
00000000005f8f60 w _ZN2v811SourceGroup13IsolateThread3RunEv
00000000005f8fb0 w _ZNSt3__113__vector_baseIPN2v86WorkerENS_9allocatorIS3_EEE5clearEv
```
But `d8.sym` also contains weird contents inside:
```
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
0000000002593fa0 P __cxa_finalize
        ...
```
The same symbol `__cxa_finalize` contains a lot of duplicated address in symbol list.
Please take a look at it. Thanks!

**Comment by honggyukim** [comment]
I think the last `__cxa_finalize` repetition may is related to #345 .

**Comment by namhyung** [comment]
Can I have the d8.sym file and the d8 binary?

**Comment by honggyukim** [comment]
Here's the symbol file.
[d8.sym.txt](https://github.com/namhyung/uftrace/files/1936521/d8.sym.txt)
Thanks!

**Comment by namhyung** [comment]
There are two problems regarding this issue.  As I said in #345 the repeated (dynamic) symbols are due to 0 entry size of the PLT section.  And as it has another mapping before the code, the base address should be calculated from the first map (instead of the code).

Please check the [check/symbol-fix](https://github.com/namhyung/uftrace/compare/check/symbol-fix) branch.

**Comment by honggyukim** [comment]
I tested it and found that it works great! The symbol problem is also fixed as well.
Thanks a lot for doing this. : )

**Comment by honggyukim** [comment]
Right, it fixes both problems including #345. Thanks.
